test/morgan.js

```diff
@@ -822,6 +822,79 @@ describe('morgan()', function () {
         test.write('0')
       })
     })
+
+    describe('a custom', function () {
+      function token (req, res, arg) {
+        return arg
+      }
+      var url
+      beforeEach(function () {
+        url = morgan.url
+        morgan.token('url', token)
+      })
+      afterEach(function () {
+        morgan.url = url
+      })
+
+      it('should overwrite existing', function (done) {
+        assert.equal(morgan.url, token)
+        done()
+      })
+
+      it('should use - if result is falsey', function (done) {
```

Comment:
I think the test description is weird here (for reference, it is: "morgan() tokens a custom should use - if result is falsey"; result of what? use it where?). Assuming the description above this is changed, this it should maybe something along the lines of "should use "-" as the token's value"--probably need to write somewhere that that is when the token returns "undefined" if you want to test falsy, you should really add more tests that guarantee that, like probably one for `false`, `null`, `0`, and `''`, which are all very interesting.

Since the interface is said to return a string, you may be able to skip the `false` and `0` I guess.